app/frontend/packs/controllers/location_autocomplete_controller.js

```diff
@@ -0,0 +1,73 @@
+import { Controller } from '@hotwired/stimulus'
+import accessibleAutocomplete from 'accessible-autocomplete'
+import debounce from 'lodash.debounce'
+import { request } from '../utils/request_helper'
+
+export default class extends Controller {
+  static targets = ['input']
+  static values = {
+    path: String,
+    minLength: { type: Number, default: 3 },
+    debounce: { type: Number, default: 200 }
+  }
+
+  connect () {
+    if (!this.hasInputTarget) return
+    this.setupAutocomplete()
+  }
+
+  setupAutocomplete () {
+    const inputElement = this.inputTarget
+
+    accessibleAutocomplete({
+      element: inputElement.parentElement,
+      id: inputElement.id,
+      showNoOptionsFound: false,
+      name: inputElement.name,
+      defaultValue: inputElement.value,
+      displayMenu: 'overlay',
+      minLength: this.minLengthValue,
+      source: this.fetchSuggestions(),
+      templates: {
+        inputValue: this.inputValueTemplate.bind(this),
+        suggestion: this.suggestionTemplate.bind(this)
+      },
+      onConfirm: this.onConfirm.bind(this),
+      confirmOnBlur: false
+    })
+
+    inputElement.remove()
+  }
+
+  fetchSuggestions () {
+    return debounce((query, populateResults) => {
+      if (!query) return populateResults([])
+
+      new Promise((resolve) => {
+        request(this.pathValue)(query, (data) => {
+          resolve(data)
+        })
+      })
+        .then((data) => {
+          const suggestions = Array.isArray(data?.suggestions) ? data.suggestions : []
```

Comment:
```suggestion
          const suggestions = [data?.suggestions].flat()
```
This will always be an Array, but will flatten it if it contains another array 👌